packages/components/pie-button/src/button.scss

```diff
@@ -1,160 +1,126 @@
-        'font-weight': dt.$font-interactive-l-weight,
-        'line-height': dt.$font-interactive-l-line-height
-    )
-);
-
-@each $size, $values in $button-sizes {
-    .o-btn--#{$size} {
-        @include button-sizing(
-            map-get($values, 'min-block-size'),
-            map-get($values, 'padding-block'),
-            map-get($values, 'padding-inline'),
-            map-get($values, 'font-size'),
-            map-get($values, 'font-weight'),
-            map-get($values, 'line-height')
-        );
+        outline: 2px solid var(--button-focus);
     }
 }
 
 // Variants
-.o-btn--primary {
-    @include button-variant($button-bg-color-primary, $button-text-color-primary);
+:host([variant='primary']) {
+    @include button-interactive-states('--dt-color-interactive-brand');
 }
 
-.o-btn--secondary {
-    @include button-variant($button-bg-color-secondary, $button-text-color-secondary);
+:host([variant='secondary']) {
+    --btn-bg-color: var(--dt-color-interactive-secondary);
+    --btn-text-color: var(--dt-color-content-interactive-secondary);
+
+    @include button-interactive-states('--dt-color-interactive-secondary');
 }
 
-.o-btn--outline {
-    @include button-variant($button-bg-color-outline, $button-text-color-outline, $button-outline-color-outline);
+:host([variant='outline']) {
+    --btn-bg-color: var(--dt-color-container-default);
+    --btn-text-color: var(--dt-color-content-interactive-tertiary);
+
+    & .o-btn {
+        border: 1px solid var(--dt-color-border-strong);
+    }
+
+    @include button-interactive-states('--dt-color-container-default');
 }
 
-.o-btn--ghost {
-    @include button-variant($button-bg-color-ghost, $button-text-color-ghost);
+:host([variant='ghost']) {
+    --btn-bg-color: var(--dt-color-container-default);
+    --btn-text-color: var(--dt-color-content-link);
+
+    @include button-interactive-states('--dt-color-container-default');
 }
 
-.o-btn.o-btn--is-disabled {
-    @include button-variant($button-bg-color-disabled, $button-text-color-disabled, $button-outline-color-disabled);
+:host([disabled]) {
+    --btn-bg-color: var(--dt-color-disabled-01);
+    --btn-text-color: var(--dt-color-content-disabled);
+
+    & .o-btn {
+        border: 1px solid var(--dt-color-disabled-01);
+        cursor: not-allowed;
+    }
+}
 
-    cursor: not-allowed;
+:host([disabled][variant='ghost']) {
+    --btn-bg-color: var(--dt-color-container-default);
 
-    &.o-btn--ghost {
-        background-color: $button-disabled-bg-color-ghost;
+    & .o-btn {
         outline: none;
     }
 }
 
 // Additional modifiers
-.o-btn--fullWidth {
+:host([isFullWidth]) .o-btn {
     inline-size: 100%;
 }
+
+:host([size='xsmall']) {
+    --btn-padding: 6px var(--dt-spacing-b);
+    --btn-min-block-size: 32px;
+    --btn-font-size: calc(var(--dt-font-size-14) * 1px);
+    --btn-line-height: calc(var(--dt-font-size-14-line-height) * 1px);
+}
+
+:host([size='small']) {
+    --btn-padding: 8px var(--dt-spacing-d);
```

Comment:
shall we use the spacing variable instead of 8px too?

Reply:
I also think it's probably worth a discussion with design around what they expect to be linked to those variables.

Mainly as the padding for the small, medium and large variants don't use the spacing 8px scale, so if the spacing changed at all in the future, then it could be that the small top/bottom padding was then bigger or smaller than the other variants as their values wouldn't adapt.

Probably a conversation to take outside of this PR potentially though?

/cc @jordifornellsJET @RLumleyJET

Reply:
Yeah seems fair to me!

Reply:
fyi, we moved away from using spacing tokens for buttons in the last update of snacks buttons for the same concern. if spacing values ever change, then our buttons will break.